### PR TITLE
Feature/cm 310 blob constructor functional test

### PR DESCRIPTION
**Pull request by euniceatcouchbase**
#### Fixes #.

- [X] Ran `flake8`
- [X] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- added test cases to cover all 3 blob constructors
- added some support for blob requests with TestServer app

### CODE REVIEW

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py [hunk 1]
```diff
@@ -3787,6 +3789,124 @@ def test_replication_behavior_with_channelRole_modification(params_from_base_tes
...
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=True,
+                                              replicator_authenticator=replicator_authenticator,
+                                              replication_type="push")
+
+    replicator.stop(repl)
+
+    # 5. Verify blob content replicated successfully
+    # sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py:3900]: u can remove this comment

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py [hunk 2]
```diff
@@ -3787,6 +3789,124 @@ def test_replication_behavior_with_channelRole_modification(params_from_base_tes
...
+        elif blob_data_type == "file_url":
+            image_file_url = blob.createImageFileUrl(image_location)
+            blob_value = blob.create("image/jpeg", file_url=image_file_url)
+
+        dictionary.setBlob(mutable_dictionary, "new_field_blob", blob_value)
+        doc_body_new = dictionary.toMap(mutable_dictionary)
+        db.updateDocument(database=cbl_db, data=doc_body_new, doc_id=doc_id)
+
+    # 4. Do push replication after docs update
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py:3891]: glad that replication is added

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py [hunk 3]
```diff
@@ -3787,6 +3789,124 @@ def test_replication_behavior_with_channelRole_modification(params_from_base_tes
...
+    dictionary = Dictionary(base_url)
+
+    doc_ids = db.getDocIds(cbl_db)
+    cbl_db_docs = db.getDocuments(cbl_db, doc_ids)
+    for doc_id, doc_body in list(cbl_db_docs.items()):
+        mutable_dictionary = dictionary.toMutableDictionary(doc_body)
+        dictionary.setString(mutable_dictionary, "new_field_string_1", random_string(length=30))
+        dictionary.setString(mutable_dictionary, "new_field_string_2", random_string(length=80))
+
+        if liteserv_platform == "android":
```
Comment (sridevi-15) [inline testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py:3862]: From line 3862 to 3888, we have to replace with utils or common method  as we might use this code repeatedly later  when we add more blob tests. you can leave it now. We can add as action item and make the change later

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py (hunk 1 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: removed

File: testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py (hunk 3 above)
Comment (euniceatcouchbase) [reply testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py]: added a new method in utils, make this code for generic use

**Pull request merged**